hibernate-core/src/test/java/org/hibernate/jpa/test/query/QueryTest.java

```diff
@@ -705,6 +705,113 @@ public void testJpaPositionalParameters() {
 		}
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HHH-12290")
+	public void testParameterCollectionAndPositional() {
+		final Item item = new Item( "Mouse", "Microsoft mouse" );
+		final Item item2 = new Item( "Computer", "Dell computer" );
+
+		EntityManager em = getOrCreateEntityManager();
```

Comment:
You could use `org.hibernate.testing.transaction.TransactionUtil2#inTransaction(org.hibernate.engine.spi.SessionFactoryImplementor, java.util.function.Consumer<org.hibernate.engine.spi.SessionImplementor>)` for this mundane transactional handling